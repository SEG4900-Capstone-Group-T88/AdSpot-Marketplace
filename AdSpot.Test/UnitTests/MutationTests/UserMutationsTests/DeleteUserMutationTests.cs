using AdSpot.Models;

namespace AdSpot.Test.UnitTests.UserMutationsTests;

[Collection("adspot-inmemory-db")]
public class DeleteUserMutationTests
{
    private const string DeleteUserMutation = """
            mutation DeleteUser($input: DeleteUserInput!) {
                deleteUser(input: $input) {
                    user {
                        userId
                        email
                        firstName
                        lastName
                    }

                    errors {
                        ... on Error {
                            __typename
                            message
                        }
                    }
                }
            }
        """;

    [Fact]
    [Trait("Category", "Unit")]
    public async Task DeleteUserSuccessful()
    {
        var result = await TestServices.ExecuteRequestAsync(
            scope =>
            {
                var context = scope.ServiceProvider.GetRequiredService<AdSpotDbContext>();
                context.Users.Add(
                    new User
                    {
                        UserId = 2,
                        Email = "testuser2@adspot.com",
                        Password = "testuserpassword2",
                        FirstName = "Test",
                        LastName = "User 2"
                    }
                );
                context.SaveChanges();
            },
            b =>
                b.SetDocument(DeleteUserMutation)
                    .SetVariableValues(
                        new Dictionary<string, object?>
                        {
                            {
                                "input",
                                new Dictionary<string, object?> { { "userId", 2 } }
                            }
                        }.AsReadOnly()
                    )
        );

        result.MatchSnapshot();
    }

    [Fact]
    [Trait("Category", "Unit")]
    public async Task DeleteUserInvalidUserIdError()
    {
        var result = await TestServices.ExecuteRequestAsync(b =>
            b.SetDocument(DeleteUserMutation)
                .SetVariableValues(
                    new Dictionary<string, object?>
                    {
                        {
                            "input",
                            new Dictionary<string, object?> { { "userId", 2 } }
                        }
                    }.AsReadOnly()
                )
        );

        result.MatchSnapshot();
    }
}
