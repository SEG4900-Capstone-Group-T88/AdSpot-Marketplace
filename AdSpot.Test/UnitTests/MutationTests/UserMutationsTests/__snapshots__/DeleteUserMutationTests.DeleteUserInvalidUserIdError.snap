{
  "data": {
    "deleteUser": {
      "user": null,
      "errors": [
        {
          "__typename": "InvalidUserIdError",
          "message": "User with id `2` does not exist."
        }
      ]
    }
  }
}